plugin/kubernetes/object/endpoint.go

```diff
@@ -56,6 +57,17 @@ func ToEndpoints(skipCleanup bool) ToFunc {
 	}
 }
 
+// SliceToEndpoints returns a function that converts an *discovery.EndpointSlice to a *Endpoints.
+func SliceToEndpoints(skipCleanup bool) ToFunc {
```

Comment:
inclined to say this should move to an `endpointslice.go` file - if these are actual objects in the k8s api, which I think they are?
function name should also be `EndpointSliceToEndpoints` to make super clear what being converrted.

Reply:
EndpointSlices _are_ separate objects, but to simplify supporting both Endpoint and EndpointSlice in the code at the same time, I am distilling both Endpoints and EndpointSlice into object.Endpoint, instead of adding a new struct called object.EndpointSlice.  Since the new conversion functions work with the existing object.Endpoints struct, I put them in `object/endpoints.go`, but I can move them to a separate file if that makes more sense.

> function name should also be EndpointSliceToEndpoints to make super clear what being converrted.

Yeah - I can change that.